chain/client/src/chunk_distribution_network.rs

```diff
@@ -0,0 +1,505 @@
+//! Helper functions for obtaining chunks from the chunk_distribution_network.
+
+use borsh::BorshDeserialize;
+use near_async::messaging::Sender;
+use near_chain::{
+    blocks_delay_tracker::BlocksDelayTracker, chain::BlockMissingChunks,
+    orphan::OrphanMissingChunks,
+};
+use near_chain_configs::{ChunkDistributionNetworkConfig, ClientConfig};
+use near_chunks::adapter::ShardsManagerRequestFromClient;
+use near_primitives::{
+    hash::CryptoHash,
+    sharding::{PartialEncodedChunk, ShardChunkHeader},
+    static_clock::StaticClock,
+    types::{EpochId, ShardId},
+};
+use std::fmt;
+use tracing::error;
+
+/// Basic interface for the chunk distribution network.
+pub trait ChunkDistributionClient {
+    type Error;
+    type Response;
+
+    async fn lookup_chunk(
+        &self,
+        prev_hash: CryptoHash,
```

Comment:
nit: could you please specify what previous hash hash corresponds to `prev_block_hash`

Reply:
Addressed as part of https://github.com/near/nearcore/pull/10480/commits/2a4d18189d23cc4a4a8d72b248ac65b8d87006e0